pandas/tests/copy_view/test_methods.py

```diff
@@ -292,6 +292,40 @@ def test_add_suffix(using_copy_on_write):
     tm.assert_frame_equal(df, df_orig)
 
 
+@pytest.mark.parametrize("axis, val", [(0, 5.5), (1, np.nan)])
+def test_dropna(using_copy_on_write, axis, val):
+    df = DataFrame({"a": [1, 2, 3], "b": [4, val, 6], "c": "d"})
+    df_orig = df.copy()
+    df2 = df.dropna(axis=axis)
+
+    if using_copy_on_write:
+        assert np.shares_memory(get_array(df2, "a"), get_array(df, "a"))
+    else:
+        assert not np.shares_memory(get_array(df2, "a"), get_array(df, "a"))
```

Comment:
You can also use `assert np.shares_memory(get_array(df2, "a"), get_array(df, "a")) is using_copy_on_write`. To me seems a bit clearer, but feel free to leave it as is if you find it more readable.

Reply:
The current one seems a bit easier to read to me and also makes it easier to identify what was hit. but not a strong opinion.

Reply:
It's also consistent with how all other tests have already been written, so if we want to change it, let's do it in a separate PR for the full file (but personally, I would also keep as is)